File: projects/engineering.purdue.edu/kak/scriptingwithobjects/swocode/chap4/WordMatch.pl
#!/usr/bin/perl -w                                                  #(A)
  
## WordMatch.pl                                                     #(B)

use strict;                                                         #(C)

use re 'eval';
#my $pattern = "hello";                                              #(D)
#my $pattern = qr/hel(?{print "first el\n"})l(?{print "second el\n"})/;
my $pattern = 'hel(?{print "first el\n"})l(?{print "second el\n"})';

print "Enter a line of text:\n";                                    #(E)
chomp( my $input_string = <> );                                     #(F)

# All outputs shown below are for the the case when the
# input string is:  "one hello is like another hello"

if ( $input_string =~ /$pattern/ ) {                                #(G)

    # The portion of the line before the match:
    print "$`\n";          # one                                    #(H)

    # The portion of the line after the match:
    print "$'\n";          #  is like another hello                 #(I)

    # The portion of the line actually matched:
    print "$&\n";          # hello                                  #(J)

    # The current line number read by <>: 
    print "$.\n";          # 1                                      #(K)

    exit 0;                                                         #(L)
}

print "no match\n";                                                 #(M)

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/WordMatch.py
#!/usr/bin/env python  
  
## WordMatch.py                                                      #(A)

import re                                                            #(B)

pattern = r'hello'                                                   #(C)

input_string = raw_input( "Enter a line of text: " )                 #(D)

m = re.search( pattern, input_string )                               #(E)

# The outputs shown below are for the case when the user enters
# in the command line the string "one hello is like another hello"
if m:                                                                #(F)
    # Print starting position index for the match:
    print m.start()                 # 4                              #(G)
    # Print the ending position index for the match:
    print m.end()                   # 9                              #(H)
    # Print a tuple of the position indices that span this match:
    print m.span()                  # (4, 9)                         #(I)
    # print the input strings characters consumed by this match:
    print m.group()                 # hello                          #(J)
else:
    print "no match"                                                 #(K)
